common/djangoapps/student/views/management.py

```diff
@@ -403,6 +404,12 @@ def change_enrollment(request, check_access=True):
         if redirect_url:
             return HttpResponse(redirect_url)
 
+        if CourseEntitlement.check_for_existing_entitlement_and_enroll(
```

Comment:
Is this why the check_for_existing_entitlement_and_enroll returns the entitlement? That seems a little off. Maybe it should just return a true/false on whether the user was enrolled? Then this wouldn't be an `is not None` check.